src/dotty/tools/dotc/core/Types.scala

```diff
@@ -869,7 +869,13 @@ object Types {
         case SkolemType(tp) =>
           tp.lookupRefined(name)
         case pre: WildcardType =>
-          WildcardType
+          pre.optBounds match {
+            case TypeBounds(lo, hi) if name.isTypeName =>
+              // for term names, forming a Termref with just the name risks losing overloaded instance
+              WildcardType(TypeBounds(NamedType(lo, name), NamedType(hi, name)))
```

Comment:
@smarter Good point about the change in lookupRefined. I have backed out of it, and replaced it with a different solution.

Test cases: We have Random.shuffle, I think that's already a good one. If you have more, please add them.